mlflow/server/js/src/experiment-tracking/components/ExperimentListView.js

```diff
@@ -90,10 +90,19 @@ export class ExperimentListView extends Component {
     this.updateSelectedExperiment('0', '');
   };
 
+  handleCheck = () => (checkedKeys, e) => {
+    if (checkedKeys.length === 0) {
+      // Do not allow un-checking all
+      e.node.checked = true;
+      return;
+    }
+    this.props.history.push(Routes.getCompareExperimentsPageRoute(checkedKeys));
+  };
```

Comment:
```suggestion
  handleCheck = (checkedKeys) => {
    if (checkedKeys.length !== 0) {
      this.props.history.push(Routes.getCompareExperimentsPageRoute(checkedKeys));
    }
  };
```

Reply:
This will shorten the code but will not implement all that I intended - if the last ticked checkbox is unticked then all the checkboxes will show unchecked but the page will not refresh, still displaying the runs from the last checked experiment. I will change the code to remove the return statement but it will have to be if...else in that case.

Reply:
Yes, that's the expected behavior, but if I omit line 96 (`e.node.checked = true;`) then you will be able to untick the last checkbox but the page will not refresh so you would be seeing all the boxes unticked but the Runs page still showing the runs from that experiment. Let me push an updated code and you tell me if it works.

Reply:
Just to be clear, I recorded this video using the suggested code.

Reply:
Sorry, my bad, I just tried that and it works. I swear it was actually unchecking the last one when I was playing with it last week. I probably had a different set of properties set for the Tree control. Will change to what you suggested 👍